internal/services/oidc.go

```diff
@@ -156,35 +155,31 @@ func (osvc *OIDCService) Decode(tokenString string) (map[string]interface{}, err
 	})
 
 	if token == nil {
-		return nil, errors.New("an unexpected error occurred while validating the JWT token")
+		return models.TokenInfo{}, errors.New("an unexpected error occurred while validating the JWT token")
 	}
 
 	if token.Valid {
-		return token.Claims.(jwt.MapClaims), nil
+		claims := token.Claims.(jwt.MapClaims)
+		return models.TokenInfo{
+			UID:  osvc.configuration.UserIDPrefix + claims[osvc.configuration.UserIDClaim].(string),
+			Name: claims[osvc.configuration.UserNameClaim].(string),
+		}, nil
 	}
 
 	if ve, ok := err.(*jwt.ValidationError); ok {
 		if ve.Errors&jwt.ValidationErrorMalformed != 0 {
-			return nil, errors.New("that's not even a token")
+			return models.TokenInfo{}, errors.New("that's not even a token")
 
 		} else if ve.Errors&jwt.ValidationErrorExpired != 0 {
-			return nil, authErrors.ErrTokenExpired
+			return models.TokenInfo{}, authErrors.ErrTokenExpired
 		} else if ve.Errors&jwt.ValidationErrorNotValidYet != 0 {
-			return nil, authErrors.ErrTokenInactiveYet
+			return models.TokenInfo{}, authErrors.ErrTokenInactiveYet
 		} else {
```

Comment:
You can remove this `else` and just `return models.TokenInfo{}, fmt.Errof(...)`.